### PR TITLE
Small cleanups on _pytest.compat

### PR DESCRIPTION
**Pull request by nicoddemus**
Small improvements and cleanups

### CODE REVIEW

File: src/_pytest/compat.py [hunk 1]
```diff
@@ -304,8 +305,8 @@ def _setup_collect_fakemodule():
 
     pytest.collect = ModuleType("pytest.collect")
     pytest.collect.__all__ = []  # used for setns
-    for attr in COLLECT_FAKEMODULE_ATTRIBUTES:
-        setattr(pytest.collect, attr, getattr(pytest, attr))
+    for attr_name in COLLECT_FAKEMODULE_ATTRIBUTES:
```
Comment (RonnyPfannschmidt) [inline src/_pytest/compat.py]: can we remove that in pytest 5?

File: src/_pytest/fixtures.py [hunk 2]
```diff
@@ -970,7 +964,7 @@ class FixtureFunctionMarker:
     name = attr.ib(default=None)
 
     def __call__(self, function):
-        if isclass(function):
+        if inspect.isclass(function):
```
Comment (RonnyPfannschmidt) [inline src/_pytest/fixtures.py]: wouldnt `isinstance(..., type)` do as we are python3 only now

File: src/_pytest/compat.py (hunk 1 above)
Comment (nicoddemus) [reply src/_pytest/compat.py]: We have this list of deprecations to remove in 5.1 (the deprecations will be errors in 5.0, removed in 5.1 as we did in 4.0/4.1).

Unfortunately it doesn't seem the `pytest.collect` has been deprecated at all, so unless I'm mistaken our only option is to deprecate it in `5.X` and remove in `6.X` I'm afraid. 😕

Comment (nicoddemus) [reply src/_pytest/compat.py]: Not sure how to implement this deprecation thought without using a `__getattr__` at module level, but that's 3.7+ only.

File: src/_pytest/compat.py [hunk 3]
```diff
@@ -129,11 +126,15 @@ def getfuncargnames(function, is_method=False, cls=None):
...
-@contextmanager
-def dummy_context_manager():
-    """Context manager that does nothing, useful in situations where you might need an actual context manager or not
-    depending on some condition. Using this allow to keep the same code"""
-    yield
+if sys.version_info < (3, 7):
+
+    @contextmanager
+    def nullcontext():
+        yield
```
Comment (asottile) [inline src/_pytest/compat.py]: coudl even do `from contextlib import ExitStack as nullcontext`

File: src/_pytest/compat.py (hunk 1 above)
Comment (asottile) [reply src/_pytest/compat.py]:
```python
class MyModule(ModuleType):
    def __getattr__(...):
        ...

sys.modules[...] = ...
```

would work (it's essentially what py37 does)

Comment (nicoddemus) [reply src/_pytest/compat.py]: Yeah I had the exact same thought right after I posted my comment, but got side-tracked by #5452. 😛

**Pull request merged**